src/datumaro/components/hl_ops/__init__.py

```diff
@@ -282,13 +282,14 @@ def merge(
         merger = get_merger(merge_policy, **kwargs)
         merged = merger(*datasets)
         env = Environment.merge(
-            dataset.env
-            for dataset in datasets
-            if hasattr(
-                dataset, "env"
-            )  # TODO: Sometimes, there is dataset which is not exactly "Dataset",
-            # e.g., VocClassificationBase. this should be fixed and every object from
-            # Dataset.import_from should have "Dataset" type.
+            [
+                dataset.env
+                for dataset in datasets
+                if hasattr(dataset, "env")
+                # TODO: Sometimes, there is dataset which is not exactly "Dataset",
+                # e.g., VocClassificationBase. this should be fixed and every object from
+                # Dataset.import_from should have "Dataset" type.
+            ]
```

Comment:
I think those does not have to convert to `list` type. Do you have any reason to set those as list?

Reply:
Besides the fact that the type hint of the merge function defines a `Sequence`, the real issue is that [here](https://github.com/williamcorsel/datumaro/blob/57f176f149c86e8061ae94b789c2d4c4724298b1/src/datumaro/components/environment.py#L272) we loop over all items to check if they're default. If you supply a generator it means that the next loop [here](https://github.com/williamcorsel/datumaro/blob/57f176f149c86e8061ae94b789c2d4c4724298b1/src/datumaro/components/environment.py#L280) is always skipped.